graph/GuidelinesGraph.md

```diff
@@ -326,6 +316,133 @@ For a complete mapping of error codes to HTTP statuses, see
+
+```diff
+<ComplexType Name="Error">
+  <Property Name="title" Type="Edm.String" />
+  <Property Name="message" Type="Edm.String" />
+  <Property Name="displayAsTip" Type="Edm.Boolean" />
++ <Property Name="displayAsAlert" Type="Edm.Boolean" />
++ <Property Name="displayAsDialog" Type="Edm.Boolean" />
+</ComplexType>
+```
+
+Additionally speaking, the workload will now also have to validate the data structure and make sure that only one of the 3 values is `true`
+
+By using an evolvable enum, instead, all we need to do is to add new members:
+
+```diff
+<ComplexType Name="Error">
+  <Property Name="title" Type="Edm.String" />
+  <Property Name="message" Type="Edm.String" />
++ <Property Name="displayMethod" Type="microsoft.graph.displayMethod" />
+-  <Property Name="displayAsTip" Type="Edm.Boolean" />
+- <Property Name="displayAsAlert" Type="Edm.Boolean" />
+- <Property Name="displayAsDialog" Type="Edm.Boolean" />
+</ComplexType>
+```
+
+```xml
+<EnumType Name="displayMethod">
+    <Member Name="tip" Value="0" />
+    <Member Name="unknownFutureValue" Value="1" />
+    <Member Name="alert" Value="2" />
+    <Member Name="dialog" Value="3" />
+</EnumType>
+```
+
+Similarly speaking, if you find yourself using a `nullable` Enum, that is a indication that maybe what you are trying to model is something that has 3 states and an enum is more appropraite. For instance, let's assume we have a boolean property called `syncEnabled`, where `null` means that the value is undefined and inherited from the general tenant configuration. Instead of modelling like a boolean:
+
+```xml
+<Property Name="syncEnabled" Type="Edm.Boolean" Nullable="true"/>
+```
+
+An enum not only better conveys the message:
+
+```xml
+<EnumType Name="syncState">
+    <Member Name="enabled" Value="0" />
+    <Member Name="disabled" Value="1" />
+    <Member Name="tenantInherit" Value="2" />
+    <Member Name="unknownFutureValue" Value="3" />
+</EnumType>
+```
+
+but it is also open for future scenarios:
+
+```diff
+<EnumType Name="syncState">
+    <Member Name="enabled" Value="0" />
+    <Member Name="disabled" Value="1" />
+    <Member Name="tenantInherit" Value="2" />
+    <Member Name="unknownFutureValue" Value="3" />
++   <Member Name="groupInherit" Value="4" />
+</EnumType>
+```
+
+Additionally speaking, depending on the situation, a nullable enum can very likely be avoided by adding a `none` member.
+
+#### Flag Enums or Collection of Enums
+
+In case an enum can have multiple values at the same time the tentation is to model the property as a collection of Enums:
+
+```xml
+<Property Name="displayMethods" Type="Collection(displayMethod)"/>
+```
+
+However, [Flagged Enums](https://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#_Toc38530378) can model this use case scenario:
+
+```diff
+- <EnumType Name="displayMethod">
++ <EnumType Name="displayMethod" isFlag="true">
+     <Member Name="tip" Value="0" />
```

Comment:
Flags enum should start at value 1.  You could argue that 0 could be added as a special value for None but it wouldn't be useful for bitwise operations.